scripts/go_release.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Helper script to prepare a Go Release for the SDKs.
+
+set -e
+
+# Read the version number and build number from the respective files
+version_number=$(< internal/version.txt)
+build_number=$(< internal/version-build.txt)
+
+# Function to validate the version number format x.y.z(-beta.w)
+validate_version_number() {
+    local version="$1"
+    if [[ "${version}" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-beta\.[0-9]+)?$ ]]; then
+        echo "Updated version number is: ${version}"
+        return 0
+    else
+        echo "Invalid version number format: ${version}"
+        return 1
+    fi
+}
+
+# Function to validate the build number format Mmmppbb
+validate_build_number() {
+    local build="$1"
+    if [[ "${build}" =~ ^[0-9]{1}[0-9]{2}[0-9]{2}[0-9]{2}$ ]]; then
+        echo "Updated build number is: ${build}"
+        return 0
+    else
+        echo "Invalid build number format: ${build}"
+        return 1
+    fi
+}
+
+# Validate the version number and build number from their respective files
+if ! validate_version_number "$version_number" || ! validate_build_number "$build_number"; then
+    exit 1
```

Comment:
We should ensure that there is a clear error message here, before exiting early. May even want to split the two cases (version/build number are invalid) for better debug ability.

Reply:
Oh wait, I am seeing this has changed - is the expectation here that the user changes the build and version number in the files before running the script? If so, we should document this, but I wonder if we have any reason for not asking for these interactively?

Reply:
I changed it as it didn't feel good having to change the file version/build numbers and then going into the script to ask the user for the version/build numbers, could increase the chance of error and if we are updating it at one spot.. its better to just grab it from there instead of asking it twice.

Reply:
You're correct - could make a lot of sense to create a script that updates these too, as well as fetches the "changelog" (release notes). e.g. `prepare-release` vs actual `release` (which does the tagging)

But I'll defer this decision to you

Reply:
As we discussed, I added the two step release process.

I added a folder with all the changelogs for the release as it was the easier way to do it than creating one changelog file and appending the newest changelogs to the top of the file as well as cutting everything out besides the most recent release changelog.